coding/python.md

```diff
@@ -7,13 +7,23 @@ This document outlines the rules for writing Python code across all our Projects
 
 ## Code style and formatting
 
-All code must conform to [PEP 8](https://pep8.org/) and be formatted with [black](https://github.com/ambv/black) using the option `--line-length 79`.
+We use [Flake8](https://flake8.pycqa.org/en/latest/) to check our code for compliance with [PEP 8](https://pep8.org/), e.g. with `flake8 webapp tests`. See [the list of Flake8 violations](https://flake8.pycqa.org/en/latest/user/error-codes.html) for more information.
 
-Project should be configured to block merging on pull requests that do not conform to these standards.
+We use [Black](https://black.readthedocs.io/en/stable/) to format our Python code using the option `--line-length 79` followed by the folders that we want to format. E.g. `black --line-length 79 webapp tests`
+
+Optionally, you can use [iSort](https://readthedocs.org/projects/isort/) to automatically rearranges your imports in different sections. It separates `import package` from `from pacakage import module` and tides up the spaces between them. However, this is not enforced by PEP8 guidelines.
+
+Project should be configured to block merging on pull requests that do not conform to these standards (currently we are using Circle CI but we are migrating to Github Actions).
 
 ## Testing
 
-Although tests are lacking on many of our existing projects, writing tests is strongly encouraged. Please write tests for any new code if at all possible, and any efforts to improve test coverage on any project would be greatly appreciated.
+Writing tests is strongly encouraged.
+
+- We write our tests in the `tests` folders. Files usually follow the `test_*.py` (pattern for unittest discovery)[https://docs.python.org/3/library/unittest.html#cmdoption-unittest-discover-p].
+- After a tests has been written, we use the `python3 -m unittest discover --start-directory tests` to run the tests.
+- Once the tests are run successfuly, you can check the coverage report by running `coverage report -m` to check that you have passed the minimum required code coverage level to avoid failing CI tests.
```

Comment:
```suggestion
- Once the tests are run successfully, you can check the coverage report by running `coverage report -m` to check that you have passed the minimum required code coverage level to avoid failing CI tests.
```